Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/StrengthViewModel.cs

```diff
@@ -1,85 +1,116 @@
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
+
 using Xamarin.Forms;
 using FindMyBLEDevice.Models;
-using System.Diagnostics;
+using System;
+using FindMyBLEDevice.Views;
 
 namespace FindMyBLEDevice.ViewModels
 {
-    [QueryProperty(nameof(BT_id), nameof(BT_id))]
+    [QueryProperty(nameof(DeviceId), nameof(DeviceId))]
     public class StrengthViewModel : BaseViewModel
     {
+        private readonly double meterScaleMin;
+        private readonly double meterScaleMax;
+                
+        private BTDevice _device = null;
         private int _radius;
-        private int _meter;
-        private string _name = "Someone's AirPods Pro";
+        private double _meter;
         private int _currentRssi;
-        private string _bt_id;
-        private int _deviceID;
-        public int Id { get; set; }
+        private int _deviceId;
+
         public StrengthViewModel()
         {
             Title = "StrengthSearch";
+            meterScaleMin = rssiToMeter(0);
+            meterScaleMax = rssiToMeter(-100);
         }
-        public string Name
+
+        public BTDevice Device
         {
-            get => _name;
-            set => SetProperty(ref _name, value);
+            get => _device;
+            set => SetProperty(ref _device, value);
         }
 
-        public string BT_id
+        public int DeviceId
         {
-            get => _bt_id;
-            set => SetProperty(ref _bt_id, value);
+            get => _deviceId;
+            set => SetProperty(ref _deviceId, value);
         }
 
         public int Radius
         {
             get => _radius;
             set => SetProperty(ref _radius, value);
         }
-        public int Meter
+
+        public double Meter
         {
             get => _meter;
-            set => SetProperty(ref _meter, value);
+            set
+            {
+                SetProperty(ref _meter, value);
+                Radius = otherScaleToRadius(meterScaleMin, meterScaleMax, Meter);
+            }
         }
 
         public int CurrentRssi
         {
             get => _currentRssi;
-            set => SetProperty(ref _currentRssi, value);
-        }
-        public int DeviceID
-        {
-            get
-            {
-                return _deviceID;
-            }
             set
             {
-                _deviceID = value;
-                LoadItemId(value);
+                SetProperty(ref _currentRssi, value);
+                Meter = rssiToMeter(CurrentRssi);
             }
         }
 
-        public async void LoadItemId(int deviceId)
+        public async void OnAppearing()
         {
-            try
-            {
-                BTDevice device = await App.DevicesStore.GetDevice(deviceId);
-                Id = device.Id;
-                Name = device.UserLabel;
-                BT_id = device.BT_GUID;
-            }
-            catch (Exception)
+            if (_deviceId != 0)
```

Comment:
This works fine, since the database assigns incrementing ids, starting from 1. 
Maybe we should add a comment, noting that? If for some reasons ids of zero were assigned, we would have problems.

Reply:
maybe initialize _deviceId with `-1` and test for `< 0`, just be sure and more clear?